libs/core/src/lib/dynamic-side-content/dynamic-side-content.component.ts

```diff
@@ -0,0 +1,151 @@
+
+let componentId = 0;
+
+@Component({
+    selector: 'fd-dynamic-side-content',
+    templateUrl: './dynamic-side-content.component.html',
+    styleUrls: ['./dynamic-side-content.component.scss'],
+    encapsulation: ViewEncapsulation.None,
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class DynamicSideContentComponent implements CssClassBuilder, OnChanges, OnInit, AfterContentInit {
+    /**
+     * Side content position, can be 'left' | 'right' | 'bottom' | 'equalSplit' | 'none'.
+     */
+    @Input()
+    set position(position: DynamicSideContentPosition) {
+        this._position = position;
+
+        this._calculateSidePosition();
+    }
+    get position(): DynamicSideContentPosition {
+        return this._position;
+    }
+
+    /**
+     * Screen size, can be 'sm' | 'md' | 'lg' | 'xl'.
+     */
+    @Input()
+    size: DynamicSideContentSize = 'xl';
+
+    /** Unique element Id, by default it's auto generated */
+    @Input()
+    @HostBinding('attr.id')
+    id = 'fd-dynamic-side-content-id-' + componentId++;
+
+    /**
+     * @hidden
+     * required by CssClassBuilder
+     */
+    class: string;
+
+    /**
+     * @hidden
+     * Indicates when side content should be rendered before or after main content
+     */
+    _isSideBefore = true;
+
+    /** @hidden */
+    @ContentChildren(DYNAMIC_SIDE_CONTENT_CHILD_TOKEN as any)
+    private _children: QueryList<DynamicSideContentMainComponent | DynamicSideContentSideComponent>;
+
+    /** @hidden */
+    private _isSideProjectedAsFirst = false;
+
+    /** @hidden */
+    private _position: DynamicSideContentPosition = 'none';
+
+    /** @hidden */
+    constructor(private _elementRef: ElementRef<HTMLElement>, private _changeDetectorRef: ChangeDetectorRef) {}
+
+    /** @hidden */
+    ngOnChanges(): void {
+        this.buildComponentCssClass();
+    }
+
+    /** @hidden */
+    ngOnInit(): void {
+        this.buildComponentCssClass();
+    }
+
+    /** @hidden */
+    ngAfterContentInit(): void {
+        this._listenToChildrenOrder();
+    }
+
+    /** @hidden */
+    @applyCssClass
+    buildComponentCssClass(): string[] {
+        return [
+            CLASS_NAME.container,
+            this.size === 'sm'
+                ? CLASS_NAME.containerSizeSm
+                : this.size === 'md' || this.size === 'lg'
+                ? CLASS_NAME.containerSizeMd
+                : this.size === 'xl'
+                ? CLASS_NAME.containerSizeXl
+                : '',
+            this._position === 'bottom' ? CLASS_NAME.containerSideBelow : '',
+            this._position === 'equalSplit' ? CLASS_NAME.containerSideEqual : ''
+        ];
+    }
+
+    /** @hidden */
+    elementRef(): ElementRef<any> {
+        return this._elementRef;
+    }
+
+    /** @hidden */
+    private _listenToChildrenOrder(): void {
+        this._children.changes.pipe(startWith(this._children)).subscribe(() => {
```

Comment:
Need to unsubscribe on destroy

Reply:
We don't need it, it [unsubscribes](https://github.com/angular/angular/blob/d1ea1f4c7f3358b730b0d94e65b00bc28cae279c/packages/core/src/linker/query_list.ts#L155) by itself